From: Andrej Shadura <andrew.shadura@collabora.co.uk>
Date: Thu, 4 Nov 2021 13:48:51 +0100
Subject: Use = with test, not == (bashism)

    checking whether to use ADOL-C hard debug mode... no
    ./configure: 19150: test: xno: unexpected operator
    ./configure: 19157: test: xno: unexpected operator
    checking whether the default adouble constructor should initialize the value to 0... yes
    ./configure: 19182: test: xyes: unexpected operator
    checking whether the adouble variables have a late initialize option for functions like malloc/realloc... no
    ./configure: 19207: test: xno: unexpected operator
    checking whether errno is thread save... no
    ./configure: 19232: test: xno: unexpected operator
---
 configure.ac | 34 +++++++++++++++++-----------------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6c2c6bb..f16fb59 100644
--- a/configure.ac
+++ b/configure.ac
@@ -29,12 +29,12 @@ AM_MAINTAINER_MODE([disable])
 
 adolclib=adolc
 
-if test "x$CFLAGS" == x ; then
+if test "x$CFLAGS" = x ; then
     ac_shell_cflags="not_set"
 else
     ac_shell_cflags="$CFLAGS"
 fi
-if test "x$CXXFLAGS" == x ; then
+if test "x$CXXFLAGS" = x ; then
     ac_shell_cxxflags="not_set"
 else
     ac_shell_cxxflags="$CXXFLAGS"
@@ -271,12 +271,12 @@ AS_HELP_STRING([--enable-harddebug],[enable ADOL-C hard debug mode
    AC_MSG_RESULT(no)
    ])
 
-if test x$adolc_harddebug == xyes ; then
+if test x$adolc_harddebug = xyes ; then
 adolc_debug=yes
 AC_DEFINE(ADOLC_HARDDEBUG,1,[ADOL-C hard debug mode])
 fi
 
-if test x$adolc_debug == xyes ; then
+if test x$adolc_debug = xyes ; then
 AC_DEFINE(ADOLC_DEBUG,1,[ADOL-C debug mode])
 fi
 
@@ -294,7 +294,7 @@ AS_HELP_STRING([--disable-stdczero],[adouble default constructor does not initia
   AC_MSG_RESULT(yes)
   ])
 
-if test x$adolc_stdczero == xyes; then
+if test x$adolc_stdczero = xyes; then
 AC_DEFINE(ADOLC_ADOUBLE_STDCZERO,1,[ADOL-C adouble zeroing mode])
 fi
 
@@ -311,7 +311,7 @@ AS_HELP_STRING([--enable-lateinit],[adouble constructors need to be called. With
   AC_MSG_RESULT(no)
   ])
 
-if test x$adolc_late_init == xyes; then
+if test x$adolc_late_init = xyes; then
 AC_DEFINE(ADOLC_ADOUBLE_LATEINIT,1,[ADOL-C adouble late initialization mode])
 fi
 
@@ -328,7 +328,7 @@ AS_HELP_STRING([--enable-tserrno],[use errno as thread number cache [default=no]
   AC_MSG_RESULT(no)
   ])
 
-if test x$adolc_tserrno == xyes ; then
+if test x$adolc_tserrno = xyes ; then
 AC_DEFINE(ADOLC_THREADSAVE_ERRNO,1,[ADOL-C thread save errno mode])
 fi
 
@@ -424,8 +424,8 @@ AC_ARG_WITH(cflags,
             [AS_HELP_STRING([--with-cflags=FLAGS],
                             [use CFLAGS=FLAGS (default: -O2)])],
             [ac_adolc_cflags="$withval"
-             if test x${adolc_debug} == xyes ||
-                test x${adolc_harddebug} == xyes ; then
+             if test x${adolc_debug} = xyes ||
+                test x${adolc_harddebug} = xyes ; then
                 ac_adolc_cflags="$ac_adolc_cflags -g -O0"
              fi
              ac_adolc_cflags="$ac_adolc_cflags $ac_adolc_openmpflag"
@@ -433,8 +433,8 @@ AC_ARG_WITH(cflags,
              AC_MSG_RESULT($ac_adolc_cflags)],
             [if test "$ac_shell_cflags" = "not_set"; then
                 ac_adolc_cflags="-O2"
-                if test x${adolc_debug} == xyes ||
-                   test x${adolc_harddebug} == xyes ; then
+                if test x${adolc_debug} = xyes ||
+                   test x${adolc_harddebug} = xyes ; then
                    ac_adolc_cflags="-g -O0 -Wall -ansi"
                 fi
              else
@@ -450,8 +450,8 @@ AC_ARG_WITH(cxxflags,
             [AS_HELP_STRING([--with-cxxflags=FLAGS],
                             [use CXXFLAGS=FLAGS (default: -O2)])],
             [ac_adolc_cxxflags="$withval"
-             if test x${adolc_debug} == xyes ||
-                test x${adolc_harddebug} == xyes ; then
+             if test x${adolc_debug} = xyes ||
+                test x${adolc_harddebug} = xyes ; then
                 ac_adolc_cxxflags="$ac_adolc_cxxflags -g -O0"
              fi
              ac_adolc_cxxflags="$ac_adolc_cxxflags $ac_adolc_openmpflag"
@@ -459,8 +459,8 @@ AC_ARG_WITH(cxxflags,
              AC_MSG_RESULT($ac_adolc_cxxflags)],
             [if test "$ac_shell_cxxflags" = "not_set"; then
                 ac_adolc_cxxflags="-O2"
-                if test x${adolc_debug} == xyes ||
-                   test x${adolc_harddebug} == xyes ; then
+                if test x${adolc_debug} = xyes ||
+                   test x${adolc_harddebug} = xyes ; then
                    ac_adolc_cxxflags="-g -O0 -Wall"
                 fi
              else
@@ -494,7 +494,7 @@ AC_ARG_ENABLE(tapedoc_values,
   ])
 AC_MSG_RESULT($tapedoc_values)
 
-if test x$tapedoc_values == xyes; then
+if test x$tapedoc_values = xyes; then
 AC_DEFINE(ADOLC_TAPE_DOC_VALUES,1,[ADOL-C tape_doc routine computes values])
 fi
 
@@ -685,7 +685,7 @@ while test "x${pdir}" != "x/" ; do
         break
     fi
 done
-if test "x${pdir}" == "x/" ; then
+if test "x${pdir}" = "x/" ; then
     writable=false
 fi
 if ! ${writable} ; then
